node/chainSimulator/process/processor.go

```diff
@@ -100,6 +92,21 @@ func (creator *blocksCreator) CreateNewBlock() error {
 	return creator.nodeHandler.GetBroadcastMessenger().BroadcastBlockDataLeader(header, miniBlocks, transactions, creator.blsKeyBytes)
 }
 
+func (creator *blocksCreator) getPreviousHeaderData() (nonce, round uint64, prevHash, prevRandSeed []byte) {
+	currentHeader := creator.nodeHandler.GetChainHandler().GetCurrentBlockHeader()
+
+	if currentHeader != nil {
+		nonce, round = currentHeader.GetNonce(), currentHeader.GetRound()
+		prevHash = creator.nodeHandler.GetChainHandler().GetCurrentBlockHeaderHash()
+		prevRandSeed = currentHeader.GetRandSeed()
+	} else {
```

Comment:
could have omitted the "else" branch and return on L101

Reply:
changed